src/user-post/service/post.service.ts

```diff
@@ -0,0 +1,188 @@
+    private readonly firebaseService: FirebaseApp,
+  ) {}
+  async createPost(
+    post: PostDto,
+    photo: Express.Multer.File,
+  ): Promise<Posts[]> {
+    const userId = await this.userRepository.findOne({
+      where: {
+        userIdToken: post.userId,
+      },
+    });
+    const newPost: Posts = this.postsRepository.create({
+      title: post.title,
+      description: post.description,
+      photo: photo ? await this.firebaseService.uploadFile(photo) : null,
+      userId: userId.id,
+    });
+    await this.postsRepository.save(newPost);
+    const createdPost: Posts[] = await this.postsRepository.find({
+      where: {
+        id: newPost.id,
+      },
+      relations: {
+        user: true,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+    });
+    return createdPost.reverse();
+  }
+  async getPosts(userId: string): Promise<Posts[]> {
+    const getUser: User = await this.userRepository
+      .findOne({
+        where: {
+          userIdToken: userId,
+        },
+      })
+      .catch(() => {
+        throw new Error('Something goes wrong');
+      });
+
+    const userPosts: Posts[] = await this.postsRepository.find({
+      where: {
+        userId: getUser.id,
+      },
+      relations: {
+        user: true,
+      },
+      take: 5,
+    });
+    return userPosts;
+  }
+  async getAllPosts(): Promise<Posts[]> {
+    const allPosts: Posts[] = await this.postsRepository.find({
+      relations: {
+        user: true,
+      },
+      take: 5,
+    });
+    return allPosts;
+  }
+  async getMorePosts(length: number): Promise<Posts[]> {
+    const morePosts: Posts[] = await this.postsRepository.find({
+      relations: {
+        user: true,
+      },
+      skip: length,
+      take: 5,
+    });
+    return morePosts;
+  }
+  async getAllPostsLength(): Promise<number> {
+    const allPosts: Posts[] = await this.postsRepository.find();
+    return allPosts.length;
+  }
+  async getAllUserPostsLength(userId: string) {
+    const getUser: User = await this.userRepository.findOneBy({
+      userIdToken: userId,
+    });
+    const allPosts: Posts[] = await this.postsRepository.find({
+      where: {
+        userId: getUser.id,
+      },
+    });
+    return allPosts.length;
+  }
+  async getMoreUserPosts(length: number, uId: string): Promise<Posts[]> {
+    console.log(length);
+    try {
+      const currentUser: User = await this.userRepository.findOne({
+        where: {
+          userIdToken: uId,
+        },
+      });
+
+      const moreUserPosts: Posts[] = await this.postsRepository.find({
+        where: {
+          userId: currentUser.id,
+        },
+        relations: {
+          user: true,
+        },
+        skip: length,
+        take: 5,
+      });
+      return moreUserPosts;
+    } catch (e) {
+      throw new Error(e.message);
+    }
+  }
+
+  async editPost(
+    editFormData: EditPostDto,
+    photo: Express.Multer.File,
```

Comment:
Should we upload it in separate place and pass only url here, or objectId makes more sense. It should be in an another service

Reply:
Not sure about another service 🤔,though i understand that my function for post editing is overcomplicated tried to fix that issue with making another function in same service that is handling part of editing suppose that will be more readable.